src/main/java/net/modfest/scatteredshards/client/screen/widget/WShardSetPanel.java

```diff
@@ -78,4 +85,17 @@ public void setShardSet(Identifier set, ShardLibraryComponent library, ShardColl
 		
 		if (host != null) this.validate(host);
 	}
+	
+	private static int shardPriority(String path) {
+		return switch(path) {
+		case "scattered_shards_visitor" -> 0;
+		case "scattered_shards_challenge" -> 1;
+		case "scattered_shards_secret" -> 2;
+		default -> Integer.MAX_VALUE;
+		};
+	}
```

Comment:
this needs to be tied to the `ShardType` instance somehow. we can't hardcode behavior like this because custom `ShardTypes` can be made

Reply:
This is a quick fix because it's not the end of the world if new shardtypes are made - they'll appear to the right of current shards in unspecified order. The eventual fix will be to allow shards to define an integer sort priority, so that shardtype creators can have control over ordering. I highly recommend this field gets added to the record *after* BC.

Reply:
Okay, reasonable.